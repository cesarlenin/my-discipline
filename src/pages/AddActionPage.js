import React, { Component } from 'react'
import ActionForm from '../components/ActionForm/ActionForm';

export class AddActionPage extends Component {
  render() {
    return (
      <section>
		<header role="banner">
			<h1>Add action</h1>
		</header>
        <section>
            <ActionForm {...this.props}/>
      </section>
      </section>
    )
  }
}

export default AddActionPage