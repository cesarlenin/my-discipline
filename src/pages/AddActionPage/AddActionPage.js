import React, { Component } from 'react'
import ActionForm from '../../components/ActionForm/ActionForm';
import BackButton from '../../components/BackButton/BackButton';

export class AddActionPage extends Component {
  render() {
    return (
      <section>
		<header role="banner">
			<h1>Add action</h1>
		</header>
    <BackButton {...this.props}/>
        <section>
            <ActionForm {...this.props}/>
      </section>
      </section>
    )
  }
}

export default AddActionPage