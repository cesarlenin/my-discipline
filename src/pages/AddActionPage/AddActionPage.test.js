import React from 'react';
import ReactDOM from 'react-dom';
import AddActionPage from './AddActionPage';
import renderer from 'react-test-renderer';

describe('<AddActionPage />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render( <
      AddActionPage / > , div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders this UI as expected', () => {
    const tree = renderer.create( <AddActionPage/> ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});