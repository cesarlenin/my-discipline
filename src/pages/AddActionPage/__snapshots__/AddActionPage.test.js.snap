// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddActionPage /> renders this UI as expected 1`] = `
<section>
  <header
    role="banner"
  >
    <h1>
      Add action
    </h1>
  </header>
  <div
    className="back"
  >
    <button
      onClick={[Function]}
    >
      Go Back
    </button>
  </div>
  <section>
    <form
      className="add-action-form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="date"
      >
        what day did you complete your habit?
      </label>
      <br />
      <input
        id="date"
        name="date"
        onChange={[Function]}
        type="date"
      />
      <br />
      <button
        disabled="date is required"
        type="submit"
        value="Submit"
      >
        Submit
      </button>
    </form>
  </section>
</section>
`;
